test/e2e/run-all.js

```diff
@@ -212,12 +213,28 @@ async function main() {
 
   console.log('My test list:', myTestList);
 
+  const changedOrNewTests = await fetchChangedE2eFiles();
+  const retriesForChangedOrNewTests = 5;
+
+  console.log('Spec files that will be re-run:', changedOrNewTests);
+
   // spawn `run-e2e-test.js` for each test in myTestList
   for (let testPath of myTestList) {
     if (testPath !== '') {
       testPath = testPath.replace('\n', ''); // sometimes there's a newline at the end of the testPath
       console.log(`\nExecuting testPath: ${testPath}\n`);
-      await runInShell('node', [...args, testPath]);
+
+      const testFileName = testPath.split('/').pop();
+      const isTestChangedOrNew = changedOrNewTests?.includes(testFileName);
+      const retryIndex = args.indexOf('--retries');
+      if (retryIndex !== -1) {
+        args.splice(retryIndex, 2);
+      }
+
+      const extraArgs = isTestChangedOrNew
+        ? ['--retry-until-failure', `--retries=${retriesForChangedOrNewTests}`]
+        : [];
```

Comment:
There's no need for this duct-taped workaround complexity.  Look at line 191 in this file.  Just put `retries` into `extraArgs` down here instead of `args` up there.

Reply:
(if you go with my other suggestion about `runningOnCircleCI` though, you can probably remove most of this anyway)

Reply:
thank you!! if we follow the approach of copying the spec files x5 times on `runningOnCircleCI` then we can indeed remove all of this logic. Just something I'm wondering, would it be okay then to treat each of those new specs the same as rest (same flags)? or we still want them to fail immediately with no-retries?

Given that @legobeat has a PR for reducing test retries, maybe it's fine to treat all the tests equally; then there's no need to add any extra flag, and just copy the spec file x times in the test list from runningOnCiircleCi, and this would simplify a lot the logic

What do you think?

Reply:
:information_source:  retries has been removed and instead we add them in the testpaths x5 times, so we benefit from parallelization.
Now only the argument for failing fast is passed here